versions.json

```diff
@@ -1,5 +1,34 @@
 {
   "versions": [
+    {
+      "path": "4.0.0-preview1/scala2.13-java17-python3-ubuntu",
+      "tags": [
+        "4.0.0-preview1-scala2.13-java17-python3-ubuntu",
+        "4.0.0-preview1-java17-python3",
+        "4.0.0-preview1-java17",
+        "python3-java17"
```

Comment:
With hindsight, I think we should have a more strict naming rule for tags. For example, `python3-java17` is a bad name as we don't even know which spark version it is.

I'd propose: `spark_version-scala_version-java_version-[python support]-[r support]-OS`

if the spark version only supports one scala or java version, we can omit the scala/java version in the name.

What do you think? @Yikun @yaooqinn

Reply:
+1 for the name. 

Is -OS necessary as a JVM app or not？If necessary, the OS versions or codenames like `Focal Fossa` also matter?

Reply:
ah we can omit OS name as well. If one day we publish docker images with different OS, we can add back the OS in the tag name.

Reply:
It's hard to believe people don't care about spark versions...

Reply:
So, the idea is  should we want to remove `python3`, `scala`, `r` special tag?

original:
```
latest: the latest version of current majar.minor version
3.3.0: the 3.3.0 version with default scala/python3
python3/r/scala: latest python3/r/scala version
3.3.0-r/3.3.0-scala/3.3.0-python3:  the 3.3.0 version with r/scala only/python3
```

tobe:
```
latest: the latest version of current majar.minor version
3.3.0: the 3.3.0 version with default scala/python3
3.3.0-r/3.3.0-scala/3.3.0-python3:  the 3.3.0 version with r/scala only/python3
```

> It's hard to believe people don't care about spark versions...

- IIRC, the background of `python3/r/scala` is that there were 3 images before (spark, spark-py, spark-r), then spark --> scala, spark-py --> python3, spark-r --> r .
- BTW, the `spark:python3`, `spark:r` is also used by https://spark.apache.org/

also cc @HyukjinKwon

Reply:
I'm wondering if we should provide a single image with all supports (Python and R). I don't think package size is a big concern here. We can also provide variants with different scala and java versions.

Reply:
yea, the all-in-one image isn't that big. Why do we provide other variants that remove python or r support?

Reply:
There are 2 reason I think:
1. To cover [kubernetes best practice](https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-how-and-why-to-build-small-container-images), to keep images size as small as possible.
2. As I said before, there were spark-py / spark-r / spark before, to align old version images.